source/Applications/Advanced/MaskPointCloud/MaskPointCloud.cpp

```diff
@@ -119,20 +123,20 @@ namespace
             }
         }
 
-        return maskedPointCloud;
+        return boost::make_shared<pcl::PointCloud<pcl::PointXYZRGB>>(maskedPointCloud);
     }
 
-    void visualizeDepthMap(const pcl::PointCloud<pcl::PointXYZRGB> &pointCloud)
+    void visualizeDepthMap(const boost::shared_ptr<pcl::PointCloud<pcl::PointXYZRGB>> &pointCloud)
```

Comment:
I think this function can continue taking the parameter as a const reference (like before).
No need for ownership semantics / manipulating lifetime.

Reply:
Then I have to call it like `visualizeDepthMap(*maskedPointCloudPCL);`

Reply:
And that makes `main `look like this:

```
        std::cout << "Converting to PCL point cloud" << std::endl;
        const auto pointCloudPCL = convertToPCLPointCloud(pointCloud);

        std::cout << "Displaying point cloud before masking" << std::endl;
        visualizePointCloud(pointCloudPCL);

        std::cout << "Displaying depth map before masking" << std::endl;
        visualizeDepthMap(*pointCloudPCL);

        std::cout << "Masking point cloud" << std::endl;
        const boost::shared_ptr<pcl::PointCloud<pcl::PointXYZRGB>> maskedPointCloudPCL =
            maskPointCloud(pointCloud, mask);

        std::cout << "Displaying point cloud after masking" << std::endl;
        visualizePointCloud(maskedPointCloudPCL);

        std::cout << "Displaying depth map after masking" << std::endl;
        visualizeDepthMap(*maskedPointCloudPCL);
```

Reply:
Do you think that is ok?

Reply:
Looks good to me 👍